### DISCUSSION TITLE
Istanbul usage with webdriverio

### DISCUSSION THREAD

**Discussion by umarkhanTR** [webdriverio/webdriverio#10925] [Technical Discussion]
Hi there

Does anyone have any good tutorial (in it's entirety) where istanbul code coverage can be integrated with E2E tests as mentioned on the website that istanbul does not require access to dev code and can do the code coverage in UI Automation test repo.

The testrunner will automatically instrument your code and and capture code coverage for you.

**Comment by christian-bromann** [comment]
Capturing test coverage for e2e test is generally not a good idea because it is quite impossible to instrument frontend and backend code. If you are interested in test coverage for individual components of your application, I recommend to look into WebdriverIOs [component testing features](https://webdriver.io/docs/component-testing) which has test coverage support out of the box.

**Reply by umer-ali-khan** [reply]
for component testing feature, I have a confusion. For component testing do I need to access the component development code or not? In component testing examples, the component code is in the same repo as test code. But in my scenario development code repository is different than UI automation code repository. Please guide.

**Reply by christian-bromann** [reply]
I am not sure why you would want to unit test components that are located outside of the project. If the project that contains the component is published somewhere e.g. on NPM than you can still test them in an external project by adding the other project as dependency.
